benches/related_entities.rs

```diff
@@ -0,0 +1,98 @@
+use bevy::prelude::*;
+use bevy_replicon::prelude::*;
+use criterion::{Criterion, criterion_group, criterion_main};
+
+criterion_main!(benches);
+
+criterion_group!(benches, hierarchy_spawning, hierarchy_changes);
+
+fn hierarchy_spawning(c: &mut Criterion) {
+    let mut group = c.benchmark_group("hierarchy_spawning");
+
+    group.bench_function("regular", |b| {
+        let mut app = App::new();
+        app.add_plugins((MinimalPlugins, RepliconPlugins));
+
+        b.iter(|| spawn_then_despawn(&mut app));
+    });
+    group.bench_function("related_without_server", |b| {
+        let mut app = App::new();
+        app.add_plugins((MinimalPlugins, RepliconPlugins))
+            .sync_related_entities::<ChildOf>();
+
+        b.iter(|| spawn_then_despawn(&mut app));
+    });
+    group.bench_function("related", |b| {
+        let mut app = App::new();
+        app.add_plugins((MinimalPlugins, RepliconPlugins))
+            .sync_related_entities::<ChildOf>();
+
+        let mut server = app.world_mut().resource_mut::<RepliconServer>();
+        server.set_running(true);
+
+        b.iter(|| spawn_then_despawn(&mut app));
+    });
+}
+
+fn spawn_then_despawn(app: &mut App) {
+    for entity in spawn_hierarchy(app.world_mut()) {
+        app.world_mut().despawn(entity);
+    }
+}
+
+fn hierarchy_changes(c: &mut Criterion) {
+    let mut group = c.benchmark_group("hierarchy_changes");
+
+    group.bench_function("regular", |b| {
+        let mut app = App::new();
+        app.add_plugins((MinimalPlugins, RepliconPlugins));
+
+        spawn_hierarchy(app.world_mut());
+
+        b.iter(|| trigger_hierarchy_change(&mut app));
+    });
+    group.bench_function("related_without_server", |b| {
+        let mut app = App::new();
+        app.add_plugins((MinimalPlugins, RepliconPlugins))
+            .sync_related_entities::<ChildOf>();
+
+        spawn_hierarchy(app.world_mut());
+
+        b.iter(|| trigger_hierarchy_change(&mut app));
+    });
+    group.bench_function("related", |b| {
+        let mut app = App::new();
+        app.add_plugins((MinimalPlugins, RepliconPlugins))
+            .sync_related_entities::<ChildOf>();
+
+        spawn_hierarchy(app.world_mut());
+
+        let mut server = app.world_mut().resource_mut::<RepliconServer>();
+        server.set_running(true);
+
+        b.iter(|| trigger_hierarchy_change(&mut app));
+    });
+}
+
+// Spawn and despawn a small hierarchy to trigger graphs rebuild.
+fn trigger_hierarchy_change(app: &mut App) {
+    app.world_mut()
+        .spawn((Replicated, children![Replicated]))
+        .despawn();
+    app.update();
```

Comment:
Can you verify that the rebuild is actually running when you do this? 0.3us seems too short.

Reply:
Yes, by locally inserting a panic in `rebuild_graphs` :sweat_smile:

Reply:
Wouldn't that trigger on the initial build, before you even get to rebuilds?

Reply:
`rebuild_graph` isn't called on every change - only once per tick before sending replication messages.
I've also ensured it triggers exclusively for the `related` benchmark when the server is running.